modules/standard/IO.chpl

```diff
@@ -5177,6 +5177,63 @@ proc _channel.writebits(v:integral, nbits:integral) throws {
   try this.write(new ioBits(v:uint(64), nbits:int(8)));
 }
 
+/*
+   Write ``numBytes`` of data from a :class:`~CTypes.c_ptr` to a ``fileWriter``
+
+   Note that native endianness is always used.
+
+   If ``numBytes`` is not evenly divisible by the size of ``t``, the remaining
+   bytes will be ignored. For example, if the ``c_ptr``'s internal type is 4
+   bytes in length, and ``numBytes=17``, only 16 bytes will be written.
+
+   .. warning::
+      This method provides no protection against attempting to access invalid memory
+
+   :arg ptr: a :class:`~CTypes.c_ptr` to some valid memory
+   :arg numBytes: the number of bytes to write
+
+   :throws SystemError: Thrown if an error occured while writing to the ``fileWriter``
+*/
+proc fileWriter.writeBinary(ptr: c_ptr(?t), numBytes: int) throws
+{
+  var e:errorCode = 0;
+  const numToWrite = numBytes / c_sizeof(t);
+
+  for i in 0..<numToWrite {
+    e = try _write_binary_internal(this._channel_internal, iokind.native, ptr[i]);
```

Comment:
See https://github.com/chapel-lang/chapel/pull/21056/files#r1035983453 -- I think it should ideally just use `qio_channel_write`.